server/szurubooru/func/posts.py

```diff
@@ -967,3 +972,81 @@ def search_by_image(image_content: bytes) -> List[Tuple[float, model.Post]]:
         ]
     else:
         return []
+
+
+PoolPostsAround = namedtuple('PoolPostsAround', 'pool first_post prev_post next_post last_post')
+
+def get_pool_posts_around(post: model.Post) -> List[PoolPostsAround]:
+    around = dict()
+    pool_ids = set()
+    post_ids = set()
+
+    dbquery = """
+    SELECT around.ord, around.pool_id, around.post_id, around.delta
+    FROM pool_post pp,
+        LATERAL get_pool_posts_around(pp.pool_id, pp.post_id) around
+    WHERE pp.post_id = :post_id;
+    """
+
+    for order, pool_id, post_id, delta in db.session.execute(dbquery, {"post_id": post.post_id}):
+        if pool_id not in around:
+            around[pool_id] = [None, None, None, None]
+        if delta == -2:
+            around[pool_id][0] = post_id
+        elif delta == -1:
+            around[pool_id][1] = post_id
+        elif delta == 1:
+            around[pool_id][2] = post_id
+        elif delta == 2:
+            around[pool_id][3] = post_id
+        pool_ids.add(pool_id)
+        post_ids.add(post_id)
+
+    pools = dict()
+    posts = dict()
+
+    for pool in db.session.query(model.Pool).filter(model.Pool.pool_id.in_(pool_ids)).all():
+        pools[pool.pool_id] = pool
+
+    for result in db.session.query(model.Post.post_id).filter(model.Post.post_id.in_(post_ids)).all():
+        post_id = result[0]
+        posts[post_id] = { "id": post_id, "thumbnailUrl": get_post_thumbnail_path_from_id(post_id) }
```

Comment:
This would be unnecessary if you let SQLAlchemy fetch the posts directly in the first place, instead of fetching the post IDs.

Also, when you do this, the `get_post_thumbnail_path_from_id()` won't be needed anymore